src/wasm-interpreter.h

```diff
@@ -1174,28 +1243,171 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     assert(flow.values.size() > curr->index);
     return Flow(flow.values[curr->index]);
   }
+  Flow visitLocalGet(LocalGet* curr) {
+    NOTE_ENTER("LocalGet");
+    NOTE_EVAL1(curr->index);
+    // Check if a constant value has been set in the context of this runner.
+    auto iter = localValues.find(curr->index);
+    if (iter != localValues.end()) {
+      return Flow(std::move(iter->second));
+    }
+    return Flow(NONCONSTANT_FLOW);
+  }
+  Flow visitLocalSet(LocalSet* curr) {
+    NOTE_ENTER("LocalSet");
+    NOTE_EVAL1(curr->index);
+    if (!(flags & FlagValues::PRESERVE_SIDEEFFECTS)) {
+      // If we are evaluating and not replacing the expression, see if there is
+      // a value flowing through a tee.
+      if (curr->type.isConcrete()) {
+        assert(curr->isTee());
+        return visit(curr->value);
+      }
+      // Otherwise remember the constant value set, if any, for subsequent gets.
+      auto setFlow = visit(curr->value);
```

Comment:
this line could be on line 1262, and then the value used instead of having two identical calls in this function.